src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -86,153 +90,139 @@ public FunctionStatus GetFunctionStatus(string name)
-                    var logData = new Dictionary<string, object>
-                    {
-                        ["Source"] = logEntry.Source,
-                        ["FunctionName"] = logEntry.FunctionName
-                    };
-                    _logger.Log(logLevel, 0, logData, null, (s, e) => logEntry.Message);
+                    traceEvent.Properties.Add(ScriptConstants.TracePropertyFunctionNameKey, logEntry.FunctionName);
                 }
 
-                return Ok();
-            }
-            else
-            {
-                return Unauthorized();
+                var logLevel = Utility.ToLogLevel(traceEvent.Level);
+                var logData = new Dictionary<string, object>
+                {
+                    ["Source"] = logEntry.Source,
+                    ["FunctionName"] = logEntry.FunctionName
+                };
+                _logger.Log(logLevel, 0, logData, null, (s, e) => logEntry.Message);
             }
+
+            return Ok();
         }
 
         [HttpPost]
         [Route("admin/host/debug")]
-        public HttpResponseMessage LaunchDebugger()
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public IActionResult LaunchDebugger()
         {
             if (_webHostSettings.IsSelfHost)
             {
                 // If debugger is already running, this will be a no-op returning true.
                 if (Debugger.Launch())
                 {
-                    return new HttpResponseMessage(HttpStatusCode.OK);
+                    return Ok();
                 }
                 else
                 {
-                    return new HttpResponseMessage(HttpStatusCode.Conflict);
+                    return StatusCode(StatusCodes.Status409Conflict);
                 }
             }
-            return new HttpResponseMessage(HttpStatusCode.NotImplemented);
+
+            return StatusCode(StatusCodes.Status501NotImplemented);
         }
 
-        public override Task<HttpResponseMessage> ExecuteAsync(HttpControllerContext controllerContext, CancellationToken cancellationToken)
+        public override void OnActionExecuting(ActionExecutingContext context)
         {
             // For all admin api requests, we'll update the ScriptHost debug timeout
             // For now, we'll enable debug mode on ANY admin requests. Since the Portal interacts through
             // the admin API this is sufficient for identifying when the Portal is connected.
             _scriptHostManager.Instance?.NotifyDebug();
 
-            return base.ExecuteAsync(controllerContext, cancellationToken);
+            base.OnActionExecuting(context);
         }
 
-        [Route("admin/extensions/{name}/{*extra}")]
         [HttpGet]
         [HttpPost]
-        [AllowAnonymous]
-        public async Task<HttpResponseMessage> ExtensionWebHookHandler(string name, CancellationToken token)
+        [Authorize(AuthenticationSchemes = AuthLevelAuthenticationDefaults.AuthenticationScheme)]
```

Comment:
This attribute declaration is confusing. I think AuthLevelAuthenticationDefaults might be misnamed? E.g. should it be just AuthenticationDefaults? Then you'd have `[Authorize(AuthenticationSchemes = AuthenticationDefaults.DefaultScheme]`

Reply:
Also, what does this annotation do exactly?

Reply:
The name is correct. This is a type with the constants/defaults for the AuthLevel authentication handler. You'll find the same pattern with the build in handlers (e.g `JwtBearerDefaults`, `MicrosoftAccountDefaults`, `CookieAuthenticationDefaults`, `OAuthDefaults`, etc.)

Reply:
The annotation specifies the required authentication scheme for this authorization.

Reply:
Ok